src/OperatingSystemApp/oscursormodel.js

```diff
@@ -0,0 +1,110 @@
+/* exported OSCursorModel */
+
+const {Gio, GLib, GObject} = imports.gi;
+const {OSModel} = imports.OperatingSystemApp.osmodel;
+
+const _propFlags = GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT;
+
+const cursorThemes = [
+    'cursor-default',
+    'cursor-cheese',
+    'cursor-ice-cream',
+    'cursor-poop',
+    'cursor-magic-wand',
+    'cursor-disco',
+    'cursor-rainbow-unicorn',
+    'cursor-glitchy',
+    'cursor-daemon',
+];
+
+var OSCursorModel = GObject.registerClass({
+    Properties: {
+        theme: GObject.ParamSpec.string(
+            'theme', 'Cursor Theme', 'cursor-default',
+            _propFlags, null),
+        size: GObject.ParamSpec.int(
+            'size', 'Cursor Size', 'Supported sizes are 16 24 32 48 64 96 128 192 256',
+            _propFlags, 16, 256, 24),
+        speed: GObject.ParamSpec.double(
+            'speed', 'Cursor Speed', '',
+            _propFlags, -1.0, 1.0, 0.0),
+    },
+}, class OSCursorModel extends OSModel {
+    _init(props = {
+        schemaId: 'org.gnome.desktop.interface',
+        schemaPath: '/org/gnome/desktop/interface/',
+    }) {
+        this._theme = null;
+        this._cursorSettings = new Gio.Settings({
+            schemaId: 'org.gnome.desktop.peripherals.touchpad',
+            path: '/org/gnome/desktop/peripherals/touchpad/',
+        });
+
+        super._init(props);
+
+        this._copiedCursors = [];
+
+        this.bindSetting('cursor-size', 'size');
+        this.bindSetting('cursor-theme', 'theme');
+        this.bindSetting('speed', 'speed', this._cursorSettings);
+    }
+
+    reset() {
+        /* Do not chain up, otherwise the theme will be reset to the default
+         * host cursor theme which does not have all the cursors sizes
+         */
+        this.theme = 'cursor-default';
+        this.settings.reset('cursor-size');
+        this._cursorSettings.reset('speed');
+    }
+
+    get theme () {
+        return this._theme;
+    }
+
+    set theme (cursor_theme) {
+        if (!cursor_theme || this._theme === cursor_theme)
+            return;
+
+        this._theme = cursorThemes.indexOf(cursor_theme) < 0 ? 'cursor-default' : cursor_theme;
+
+        if (this._copiedCursors.indexOf(this._theme) >= 0) {
+            this.notify('theme');
+            return;
+        }
+
+        /* Build cursor resource path */
+        var cursor = Gio.File.new_for_uri(
+            `resource:///com/endlessm/HackToolbox/OperatingSystemApp/cursors/${
+                this._theme}.xmc`);
+
+        /* Build user custom cursors directory path */
+        var dir = Gio.File.new_for_path(GLib.build_pathv('/',
+            [GLib.get_home_dir(), '.icons', this._theme, 'cursors']));
```

Comment:
Try to see if `~/.local/share/icons` works too (i.e. `GLib.get_user_data_dir()/share/icons`)

Reply:
(It doesn't. libXCursor which is what is used to load cursor themes does not look in there, unfortunately.)